cw-orch-daemon/src/state.rs

```diff
@@ -9,25 +9,51 @@ use cw_orch_core::{
     CwEnvError, CwOrchEnvVars,
 };
 use ibc_chain_registry::chain::ChainData;
+use once_cell::sync::Lazy;
 use serde::Serialize;
 use serde_json::{json, Value};
-use std::{collections::HashMap, fs::File, path::Path};
+use std::{collections::HashMap, path::Path, sync::Mutex};
 use tonic::transport::Channel;
 
+pub(crate) static GLOBAL_WRITE_STATE: Lazy<Mutex<HashMap<String, (u64, JsonFileState)>>> =
+    Lazy::new(|| Mutex::new(HashMap::new()));
+
 /// Stores the chain information and deployment state.
 /// Uses a simple JSON file to store the deployment information locally.
-#[derive(Clone, Debug)]
+#[derive(Debug)]
 pub struct DaemonState {
     /// this is passed via env var STATE_FILE
-    pub json_file_path: String,
+    json_file_path: String,
     /// Deployment identifier
     pub deployment_id: String,
     /// gRPC channel
     pub grpc_channel: Channel,
     /// Information about the chain
     pub chain_data: ChainData,
     /// Flag to set the daemon state readonly and not pollute the env file
-    pub read_only: bool,
+    read_only: bool,
```

Comment:
When would this flag be used? Could you elaborate on the comment? What is meant with not polluting the env file?

Reply:
Yes this is used for clone-testing to make sure the dameon is not writing to the `state` file in the clone-testing case, this is a mistake here.